lib/features/map_view/widgets/bottom_scroll_sheet/map_view_pop_behaviour.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../../controllers/bottom_sheet_controller.dart';
+import '../../controllers/controllers_set.dart';
+import '../map_config.dart';
+
+class MapViewPopBehaviour<T extends GoogleNavigable> extends ConsumerWidget {
+  const MapViewPopBehaviour({
+    required this.child,
+    required this.screenHeight,
+    super.key,
+  });
+  final double screenHeight;
+  final Widget child;
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isAlmostFullyExtended =
+        ref.watch(bottomSheetPixelsProvider) / screenHeight > 0.98;
+
+    final isAnyActive = ref.watch(context.activeMarkerController<T>()) != null;
+
+    return PopScope(
+        canPop: !isAnyActive && !isAlmostFullyExtended,
+        onPopInvoked: (didPop) {
+          if (!didPop) {
+            ref.read(context.mapController<T>().notifier).onMapBgTap(null);
```

Comment:
I'd consider renaming the `onMapBgTap `to sth more meaningfull like `onMapBackgroundTap`. I took me a while to understand what it exactly does. But maybe it's just me 😝😝

Reply:
Yeah, that's actually very good note. it was obviously obvious to me, but it may not be for all people